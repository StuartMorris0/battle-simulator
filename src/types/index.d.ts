declare interface PlayerHealth {
  current: number;
  lost: number | undefined;
}

declare interface Player {
  playerName: string;
  dice1: number;
  dice2: number;
  health: PlayerHealth;
  hasWon: boolean;
}
