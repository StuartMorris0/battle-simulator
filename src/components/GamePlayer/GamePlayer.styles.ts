import { motion } from 'framer-motion';
import styled from 'styled-components';

export const GamePlayerWrapper = styled(motion.div)`
  display: flex;
  justify-content: space-evenly;
  flex-grow: 0.5;
  align-items: center;
  height: 100%;
  overflow: hidden;
`;

export const DiceHolder = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  padding: 12px;
  height: 120px;
  width: 80px;
`;
