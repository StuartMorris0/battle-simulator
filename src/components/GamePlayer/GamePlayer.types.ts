export interface IGamePlayerProps {
  player: Player;
  isRolling: boolean;
}
