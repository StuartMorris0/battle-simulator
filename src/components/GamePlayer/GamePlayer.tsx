import React from 'react';

import Dice from 'components/Dice';
import HealthBar from 'components/HealthBar';
import PlayerAvatar from 'components/PlayerAvatar';

import { DiceHolder, GamePlayerWrapper } from './GamePlayer.styles';
import { IGamePlayerProps } from './GamePlayer.types';

const GamePlayer: React.FC<IGamePlayerProps> = ({ player, isRolling }: IGamePlayerProps) => {
  const {
    playerName,
    dice1,
    dice2,
    health: { current, lost },
    hasWon,
  } = player;
  return (
    <GamePlayerWrapper data-testid="GamePlayer" animate={{ scale: 1 }} className="player">
      <PlayerAvatar name={playerName} hasWon={hasWon} />
      <DiceHolder>
        <Dice value={dice1} isRolling={isRolling} />
        <Dice value={dice2} isRolling={isRolling} />
      </DiceHolder>
      <HealthBar current={current} lost={lost} />
    </GamePlayerWrapper>
  );
};

export default GamePlayer;
