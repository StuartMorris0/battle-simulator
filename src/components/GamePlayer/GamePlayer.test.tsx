import React from 'react';

import { render, screen } from 'global/utils/test-utils';

import GamePlayer from '.';
import { IGamePlayerProps } from './GamePlayer.types';

describe('GamePlayer Component', () => {
  let props: IGamePlayerProps;
  beforeEach(() => {
    props = {
      player: {
        playerName: 'PLAYER',
        dice1: 1,
        dice2: 2,
        health: {
          current: 85,
          lost: 1,
        },
        hasWon: false,
      },
    };
  });

  const renderComponent = () => render(<GamePlayer {...props} />);

  test('renders component', () => {
    const { getByTestId, getByText } = renderComponent();
    expect(getByTestId('GamePlayer')).toBeInTheDocument();
    expect(getByText('PLAYER')).toBeInTheDocument();
    expect(getByText('85')).toBeInTheDocument();
  });

  test('render snaphost', () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
