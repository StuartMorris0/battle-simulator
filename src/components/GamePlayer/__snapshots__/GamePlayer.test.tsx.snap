// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePlayer Component render snaphost 1`] = `
.c4 {
  display: block;
}

.c4 img {
  width: 50px;
  height: 50px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c6 {
  position: absolute;
  top: 50%;
  left: 25%;
  background: white;
  border-radius: 26px;
  padding: 10px;
  border: 1px solid #f1f1f1;
  font-size: 20px;
  font-weight: bold;
}

.c9 {
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  font-weight: bold;
}

.c7 {
  background-color: white;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 3px;
  width: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 2px;
}

.c8 {
  background-color: #51da51;
  border-radius: 4px;
  display: block;
  width: 100%;
  height: 85%;
  -webkit-transition: height 1s ease-in-out;
  transition: height 1s ease-in-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  padding: 10px;
  position: relative;
}

.c2 {
  width: 50px;
  margin: 12px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-box-flex: 0.5;
  -webkit-flex-grow: 0.5;
  -ms-flex-positive: 0.5;
  flex-grow: 0.5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  overflow: hidden;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  padding: 12px;
  height: 120px;
  width: 80px;
}

@media (min-width:769px) {
  .c5 {
    height: 260px;
  }
}

@media (min-width:769px) {
  .c2 {
    width: 80px;
  }
}

<div
  class="c0 player"
  data-testid="GamePlayer"
>
  <div
    class="c1"
    data-testid="PlayerAvatar"
  >
    <img
      alt="PLAYER"
      class="c2"
      src="player.svg"
    />
    <span>
      PLAYER
    </span>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4"
      data-testid="RolledDice"
    >
      <img
        alt="dice"
        src="one.svg"
      />
    </div>
    <div
      class="c4"
      data-testid="RolledDice"
    >
      <img
        alt="dice"
        src="two.svg"
      />
    </div>
  </div>
  <div
    class="c5"
    data-testid="HealthBar"
  >
    <div
      class="c6"
      style="opacity: 0; transform: none;"
    >
      -1
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      />
    </div>
    <div
      class="c9"
    >
      85
    </div>
  </div>
</div>
`;
